frontend/src/components/pages/Scheduling/SelectDateTime.tsx

```diff
@@ -9,39 +24,247 @@ const SelectDateTime = ({
-  isBeingEdited,
 }: SchedulingStepProps) => {
   const { previous, next } = navigation;
-  const { dayPart } = formValues;
+  const {
+    dayPart,
+    frequency,
+    startTime,
+    endTime,
+    recurringDonationEndDate,
+  } = formValues;
+  const { authenticatedUser } = useContext(AuthContext);
+
+  enum DayParts {
+    EARLY_MORNING = "Early Morning (12am - 6am)",
+    MORNING = "Morning (6am - 11am)",
+    AFTERNOON = "Afternoon (11am - 4pm)",
+    EVENING = "Evening (4pm - 9pm)",
+    NIGHT = "Night (9pm - 12am)",
+  }
 
   const dayParts = [
-    "Early Morning (12am - 6am)",
-    "Morning (6am - 11am)",
-    "Afternoon (11pm - 4pm)",
-    "Evening (4pm - 9pm)",
-    "Night (9pm - 12am)",
+    DayParts.EARLY_MORNING,
+    DayParts.MORNING,
+    DayParts.AFTERNOON,
+    DayParts.EVENING,
+    DayParts.NIGHT,
   ];
 
+  const timeRanges = {
+    earlyMorning: [
+      "12:00 AM - 1:00AM",
+      "1:00 AM - 2:00AM",
+      "2:00 AM - 3:00AM",
+      "3:00 AM - 4:00AM",
+      "4:00 AM - 5:00AM",
+      "5:00 AM - 6:00AM",
+    ],
+    morning: [
+      "6:00 AM - 7:00 AM",
+      "7:00 AM - 8:00 AM",
+      "8:00 AM - 9:00 AM",
+      "9:00 AM - 10:00 AM",
+      "10:00 AM - 11:00 AM",
+    ],
+    afternoon: [
+      "11:00 AM - 12:00 PM",
+      "12:00 PM - 1:00 PM",
+      "1:00 PM - 2:00 PM",
+      "2:00 PM - 3:00 PM",
+      "3:00 PM - 4:00 PM",
+    ],
+    evening: [
+      "4:00 PM - 5:00 PM",
+      "5:00 PM - 6:00 PM",
+      "6:00 PM - 7:00 PM",
+      "7:00 PM - 8:00 PM",
+      "8:00 PM - 9:00 PM",
+    ],
+    night: ["9:00 PM - 10:00 PM", "10:00 PM - 11:00 PM", "11:00 PM - 12:00 AM"],
+  };
+
+  const frequencies = ["One time donation", "Daily", "Weekly", "Monthly"];
+
+  const getTimeSlot = (selectedDayPart: string) => {
+    switch (selectedDayPart) {
+      case DayParts.EARLY_MORNING:
+        return timeRanges.earlyMorning;
+      case DayParts.MORNING:
+        return timeRanges.morning;
+      case DayParts.AFTERNOON:
+        return timeRanges.afternoon;
+      case DayParts.EVENING:
+        return timeRanges.evening;
+      case DayParts.NIGHT:
+        return timeRanges.night;
+      default:
+        return null;
+    }
+  };
+
+  const get12HTimeString = (type: "start" | "end") => {
+    const time = type === "start" ? startTime : endTime;
+    const time24Hour = `${new Date(time).getHours().toString()}:00`;
+    return moment(time24Hour, "HH:mm").format("h:mm A");
+  };
+
+  // setting state initial values
+  const [date, setDate] = useState<Date>(new Date(startTime));
+  const [timeRange, setTimeRange] = useState(
+    `${get12HTimeString("start")} - ${get12HTimeString("end")}`,
+  );
+  const [showTimeSlots, setShowTimeSlots] = useState<string[] | null>(
+    getTimeSlot(dayPart),
+  );
+  const [schedules, setSchedules] = useState<Schedule[]>([]);
+  // TODO: Need to edit this to update + calculate based on schedules pulled from backend
+  const [icons, setIcons] = useState<number[]>([0, 0, 1, 0, 3]);
```

Comment:
are we going to have it manually be the default # for this PR?

Reply:
Yup, styling the calendar + doing the person icons will be in a separate PR